apps/web/src/components/Composer/Actions/Attachment.tsx

```diff
@@ -22,33 +22,20 @@ const Attachment: FC = () => {
   const isUploading = usePublicationStore((state) => state.isUploading);
   const { handleUploadAttachments } = useUploadAttachments();
   const [showMenu, setShowMenu] = useState(false);
+  const [limitReached, toggleLimitReached] = useState(false);
+  const [isImageAttachmentType, setIsImageAttachmentType] = useState(false);
   const id = useId();
   const dropdownRef = useRef(null);
 
-  useOnClickOutside(dropdownRef, () => setShowMenu(false));
-
-  const hasVideos = (files: FileList) => {
-    let videos = 0;
-    let images = 0;
-
-    for (const file of files) {
-      if (ALLOWED_VIDEO_TYPES.includes(file.type)) {
-        videos = videos + 1;
-      } else {
-        images = images + 1;
-      }
-    }
-
-    if (videos > 0) {
-      if (videos > 1) {
-        return true;
-      }
-
-      return images > 0 ? true : false;
+  useEffect(() => {
+    //Clear media upload limits if attachments are empty
+    if (!attachments.length) {
+      toggleLimitReached(false);
+      setIsImageAttachmentType(false);
     }
+  }, [attachments]);
```

Comment:
Can we avoid useEffect?

https://react.dev/learn/you-might-not-need-an-effect

Reply:
cc @harish-sethuraman 🙇🏼

Reply:
@bigint I experimented with all the methods in the doc you sent

Our options are, move all the logic up to parent, then pass down to Attachment with Key, this would complicate things and expose the Attachment component to props it doesn't need.

I looked into setting up a listener for the store data, but that doesn't seem to possible.

I believe useEffect here makes the most sense, but if you insist on no useEffect I will look into changing it

Reply:
Resolved as following, thanks for the resources:
https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes